airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -790,13 +777,16 @@ public void testSyncAfterUpgradeToPerStreamState(final TestInfo testInfo) throws
         STREAM_NAME));
 
     // sync one more time. verify it is the equivalent of a full refresh.
+    final String expectedState = "{\"cdc\":false,\"streams\":[{\"cursor\":\"6\",\"stream_name\":\"id_and_name\",\"cursor_field\":[\"id\"],\"stream_namespace\":\"public\"}]}";
     LOGGER.info("Starting {} sync 3", testInfo.getDisplayName());
     final JobInfoRead connectionSyncRead3 =
         apiClient.getConnectionApi().syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
     waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead3.getJob());
-    LOGGER.info("state after sync 3: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+    final ConnectionState state = apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId));
```

Comment:
My comment was more about having a check on the state type, like using the new repository done by Jimmy and check that it is legacy here. WDYT?

Reply:
Yeah I think it would be useful if in every place in this test where we are currently calling and just logging `apiClient.getConnectionApi().getState()`, we instead explicitly check that the `ConnectionState` object that was returned has the expected `stateType` and the expected state/globalState/streamState field is set to the expected value. 

E.g. I think after sync 1, we expect it to be a `LEGACY` state. After sync 2, we expect a `STREAM` state. After the reset, we expect a `NOT_SET` state. And after sync 3, we expect a `STREAM` state again.

Reply:
Yeah, That sounds closer to the acceptance test spirit to use the API to do any operation

Reply:
> E.g. I think after sync 1, we expect it to be a LEGACY state. After sync 2, we expect a STREAM state. After the reset, we expect a NOT_SET state. And after sync 3, we expect a STREAM state again.

This will depend on our ability to control the feature flag (Passing it to the container that have the flag). This is what I am investigating at the moment but for now it should always be `LEGACY`.

Reply:
@benmoriceau I think we can do that as a separate PR once everything is ready for use.  For now, this is verifying that the existing state stuff works even with the updated connector.